File: 0868-binary-gap/0868-binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        res = 0
        mask = 1
        prev = -1
        for i in range(32):
            if mask & n:
                if prev != -1:
                    res = max(res, i - prev)
                prev = i
            mask <<= 1
        return res

File: 0868-binary-gap/0868-binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        int res = 0;
        int mask = 1;
        char prev = -1;
        for(int i = 0; i < 32; i++)
        {
            if(mask & n)
            {
                if(prev != -1)
                {
                    res = max(res, i - prev);
                }
                prev = i;
            }
            mask <<= 1;
        }
        return res;
    }
};